tests/unittests/benchmark/test_benchmark_client.py

```diff
@@ -306,3 +307,60 @@ def insert_race_condition(*args, **kwargs):
 
             assert bm.configuration == benchmark_config
             assert count_benchmarks() == 1
+
+    def test_create_with_executor(self, benchmark_config, benchmark_config_py):
+
+        with OrionState():
+            config = copy.deepcopy(benchmark_config_py)
+            bm1 = get_or_create_benchmark(**config)
+
+            assert bm1.configuration == benchmark_config
+            assert bm1.executor.n_workers == orion.core.config.worker.n_workers
+
+            executor = Joblib(n_workers=2, backend="threading")
+            config["executor"] = executor
+            bm2 = get_or_create_benchmark(**config)
+
+            assert bm2.configuration == benchmark_config
```

Comment:
Do you think executor should be part of the config? If this affects parallelism and results it may be better to include it.

Reply:
I thought this a little bit. I took it out for 2 consideration:
1) this configure is primary useful when user run the one benchmark on multiple workers(manually create benchmark separately and launch it). It should be okay that at different worker use different executor configure.
2) this configure will be persisted, and now we pass-in a executor object directly instead of the executor configure, to do the persist, need to change this or ask executor return a configure for persistence.

Reply:
Do you think we would need to persist the configuration in the case of the parallelization assessment? I seems to me it would be better to. Maybe that can be another PR, since it would require supporting configuration of the executor through dict arguments instead of simply an instantiated object.

Reply:
Like here: https://github.com/Epistimio/orion/pull/634/files/cb4edf3667330ad6863cfde698f7591b91a281ed..5032cc1fa5c1323d65f6aa0c405530ac9816f617#diff-850efe1c00bd8d6203e80177ccbf6bcc12058cb1d15c8d56b8902728379bf94cR313

Reply:
For parallelization assessment, yes, I think we need to pass-in executor name and configure instead of an instance for persistence, this is because it is part of the benchmark and should not change when we launch same benchmark at different nodes.